qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/VisitNotePage.java

```diff
@@ -119,10 +120,12 @@ public void confirmDeleteButton() {
 	}
 
 	public String primaryDiagnosis() {
+		waiter.until(ExpectedConditions.visibilityOfElementLocated(PRIMARY_DIAGNOSIS_ELEMENT));
 		return findElement(PRIMARY_DIAGNOSIS_ELEMENT).getText().trim();
 	}
 
 	public String secondaryDiagnosis() {
+		waiter.until(ExpectedConditions.visibilityOfElementLocated(SECONDARY_DIAGNOSIS_ELEMENT));
```

Comment:
Basically, this was to help us confirm that using this method anywhere will return error with waiting visibility issue when we directly use it to check for an assertion., this will break not until i use direct check using assert text.